Improve bank letter

ElGigi opened the pull request:
## Translations

To allow user to pass translations, i added one argument in constructor to pass translations.

## Render

- Add argument in constructor to pass CSS styles
- Improve default render of HTML letter

Preview:

![HTML](https://user-images.githubusercontent.com/18268216/180181682-b76fad0c-ee92-4980-9963-f13e3914ccb8.png)

Print preview with page break:

![Print](https://user-images.githubusercontent.com/18268216/180181677-e14c3165-141f-4ab0-810c-f6cef1f0bc07.png)

ElGigi committed:
Improve bank letter

andrew-svirin commented on src/Services/BankLetter/Formatter/HtmlBankLetterFormatter.php:
```diff
@@ -21,165 +18,187 @@
+    private $style;
 
-    public function format(BankLetter $bankLetter)
+    public function __construct(array $translations = [], string $style = '')
```
1. Optional arguments better will llok when work by setters, so instead to pass into constructor - add 2 setter `setTranslation` & `setStyle`
2. Constructor should do nothing except assign required parameters. If it needs to have some default values, when specified method can be implemented.

andrew-svirin commented on src/Services/BankLetter/Formatter/HtmlBankLetterFormatter.php:
```diff
@@ -21,165 +18,187 @@
+
+    public function format(BankLetter $bankLetter): string
+    {
+        if (empty($serverName = $bankLetter->getBank()->getServerName())) {
```
Should be extracted to separated method to follow SRP.

ElGigi committed:
Merge branch '2.x' into bank-letter

# Conflicts:
#	src/Services/BankLetter/Formatter/HtmlBankLetterFormatter.php

ElGigi committed:
New methods:

- HtmlBankLetterFormatter::setTranslations()
- HtmlBankLetterFormatter::setStyle()
- PdfBankLetterFormatter::setTranslations()
- PdfBankLetterFormatter::setStyle()

ElGigi committed:
SRP

ElGigi committed:
Merge branch '2.x' into bank-letter

andrew-svirin commented:
Please check this line https://github.com/andrew-svirin/ebics-client-php/pull/167/files#diff-b0074cd6930886a2c756ae6c99746363544aab3eff635e9ac2544eb7a5a38a7cR1264

ElGigi commented:
> @ElGigi Please check this line https://github.com/andrew-svirin/ebics-client-php/pull/167/files#diff-b0074cd6930886a2c756ae6c99746363544aab3eff635e9ac2544eb7a5a38a7cR1264

So good 👍
